modules/gdpr_fields/src/Controller/GDPRController.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+namespace Drupal\gdpr_fields\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\gdpr_fields\GDPRCollector;
+use Drupal\user\UserInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Returns responses for GDPR Field routes.
+ */
+class GDPRController extends ControllerBase {
+
+  /**
+   * Stores the Views data cache object.
+   *
+   * @var \Drupal\gdpr_fields\GDPRCollector
+   */
+  protected $collector;
+
+  /**
+   * Constructs a new GDPRController.
+   *
+   * @param \Drupal\gdpr_fields\GDPRCollector $collector
+   *   The GDPR collector service.
+   */
+  public function __construct(GDPRCollector $collector) {
+    $this->collector = $collector;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('gdpr_fields.collector')
+    );
+  }
+
+  /**
+   * Lists all fields with GDPR sensitivity.
+   *
+   * @return array
+   *   The Views plugins report page.
+   */
+  public function fieldsList($mode) {
+    $output = [];
+    $entities = [];
+    $include_not_configured = $mode == 'all';
+    $this->collector->getEntities($entities);
```

Comment:
This will return only user fields as the second parameter has 'user' as the default value.

Reply:
GDPRCollector::getEntities() This is a self referencing function. User is the default entity when no other option is provided because we almost always want to start with the user. However it calls itself with other entities that reference the user and thus will add other entities as well.

Eg `$this->getEntities($entity_list, $entity_type, $bundle_id);`
    > GDPRCollector::getEntities():70